### ISSUE TITLE
[Q]: Padding without masking

### ISSUE THREAD

**Issue by ariG23498** [carted/handling-variable-length-text-tf#12]
_Note: This is a question and not an issue (could not seem to tag this as one)._

Hey folks,

Loved the implementation of [handling variable-length text](https://github.com/carted/handling-variable-length-text-tf/blob/main/smart-batching-shallow-mlp.ipynb).

With the [`padded_batch` ](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#padded_batch) API, the dynamic batching and padding is really commendable. I was thinking whether there should be a procedure to mask the padding as well in the model.

I mostly go with `mask_zero=True` of the [`layers.Embedding`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Embedding) API.

Would that be an approach you would consider? If not, what could be a better or any other alternative to handling the situation.

**Comment by sayakpaul** [comment]
Yeah, I think setting `mask_zero=True` should cut it. If you're using any value other than 0 (padding value for `layers.Embedding`), then you'd need to implement a custom embedding layer to denote this change.
